code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/xdm/XDMAdvertisingDetails.kt

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.media.internal.xdm
+
+internal data class XDMAdvertisingDetails(
+    var friendlyName: String? = null,
```

Comment:
Do we need to assign null for optionals?

Reply:
The properties need default values otherwise they'll need to be passed to the default constructor. For example, `XDMAdvertisingDetails` is created by `val xdm = XDMAdvertisingDetails()` vs. needing to pass all 11 properties.